StandardLibraryEssentials/StandardTemplateLibrary/IteratorCategories/task.md

```diff
@@ -0,0 +1,60 @@
+Sometimes you will need to create your own iterators. For example, you may want to create an iterator that iterates over a custom data structure. In such cases, you will need to know how to create iterators that conform to the standard iterator categories. Different iterator categories are expected in different situations. For example, the algorithms in the STL require iterators that support the forward iterator category. If you want to use an algorithm that requires forward iterators, you will need to ensure that your custom iterator supports the forward iterator category.
```

Comment:
Let's remove the part about "creating your own iterators". Is is quite an advanced topic. Let's focus on iterator categories from the user point of view, and describe what category of iterators is provided by each discussed STL container.